service/storage_service.cc

```diff
@@ -831,6 +831,21 @@ storage_service::get_range_to_address_map(const sstring& keyspace,
     return construct_range_to_endpoint_map(ks, get_all_ranges(sorted_tokens));
 }
 
+void storage_service::handle_state_replacing_update_pending_ranges(inet_address replacing_node) {
+    try {
+        slogger.info("handle_state_replacing: Waiting for replacing node {} to be alive on all shards", replacing_node);
+        _gossiper.wait_alive({replacing_node}, std::chrono::milliseconds(5 * 1000));
```

Comment:
I see. Maybe it would be simpler to update is_alive on shard 0 last, so that listeners like this could assume that all shards are already updated. No need to change now.